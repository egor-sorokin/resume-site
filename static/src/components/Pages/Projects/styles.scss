@import '../../styles/base/_vars.scss';

.slide {
  min-height: 100vh;
}

.container {
  position: relative;

  display: flex;
  flex-direction: row;

  padding: 6vh 2vw 16vh;
}

.content {
  width: 50%;

  &--left {
    padding-left: 6vw;
  }
  &--right {
    padding-right: 6vw;
  }
}

/*
.logo {
  position: absolute;
  left: 6%;
  top: 12%;

  width: 180px;
  height: 180px;
}
*/

.tiles {
  display: flex;
  flex-direction: column;
}

.tile {
  width: 330px;
  min-height: 330px;
  height: 100%;

  padding: 20px 40px;

  background-color: $dune;
  box-shadow: -4px 8px 10px 1px transparentize($dune, 0.5);

  color: $mercury;

  &:nth-child(2n+1) {
    align-self: flex-end;

    .tile__text {
      margin-top: 30px;
    }
  }
  &:nth-child(2n+2) {
    align-self: flex-start;

    margin-top: 5vh;

    .tile__text {
      margin-top: 10px;

      opacity: 0.9;

      font-family: $secondaryFont;
      font-size: 14px;
      font-weight: 300;
      line-height: 1.65;
      letter-spacing: 0.3px;
    }
  }
  &__title {
    font-size: 26px;
  }
  &__subtitle {
    font-family: $secondaryFont;
    font-size: 12px;
    text-transform: uppercase;
    letter-spacing: 2px;
  }
}

.link {
  display: inline-block;
  position: relative;

  cursor: pointer;

  &__placeholder {
    color: $mercuryLight;
  }
  &__mask {
    overflow: hidden;

    position: absolute;
    top: 0;
    left: 0;

    transform: translateX(-100%) translateZ(0);
    transition: transform $defaultTiming $defaultEasing;
  }
  &__text {
    display: block;

    color: $mercury;

    transform: translateX(100%) translateZ(0);
    transition: transform $defaultTiming $defaultEasing;
  }
}

.link:hover {
  .link__mask {
    transform: translateX(0) translateZ(0)
  }
  .link__text {
    transform: translateX(0) translateZ(0);
  }
}

.content--reverse {
  .logo {
    right: 6%;
    left: initial;
  }

  .tile {
    box-shadow: 4px 8px 10px 1px transparentize($dune, 0.5);

    color: $mercury;

    &:nth-child(2n+1) {
      align-self: flex-start;

      margin-top: 5vh;
    }
    &:nth-child(2n+2) {
      align-self: flex-end;
    }
  }
}
