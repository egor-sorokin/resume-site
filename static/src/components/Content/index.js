import React from 'react';
import PropTypes from 'prop-types';

import DescriptionList from '../DescriptionList/index';
import ImageGrid from '../ImageGrid/index';
import './styles.scss';


const Content = ({
  name, info, image, experiments,
}) => (
  <section className="project">
    <div className="col col--left">
      <h1 className="font-s-40">
        {name}
      </h1>
      <DescriptionList
        listData={info}
      />
    </div>
    <div className="col col--right">
      {experiments.length === 0 ? (
        <img
          className="image"
          title={image.alt}
          alt={image.alt}
          src={image.src}
        />
      ) : (
        <ImageGrid
          gridData={experiments}
        />
      )}
    </div>
  </section>
);

Content.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.shape({
    alt: PropTypes.string,
    src: PropTypes.string,
  }),
  info: PropTypes.instanceOf(Array),
  experiments: PropTypes.instanceOf(Array),
};

Content.defaultProps = {
  image: {
    alt: '',
    src: '',
  },
  info: [],
  experiments: [],
};

export default Content;
