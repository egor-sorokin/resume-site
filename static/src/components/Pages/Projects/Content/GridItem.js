import React from 'react';
import './styles.scss';


const GridItem = ({ url }) => 
  <li className="item">
    <a href={url}></a>
  </li>;

export default GridItem;
