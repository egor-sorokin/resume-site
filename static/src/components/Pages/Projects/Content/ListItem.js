import React from 'react';
import './styles.scss';


const ListItem = ({ title, text, url }) => 
  <li className="item">
    <h6 className="item__title">{title}</h6>
      {url ? (
        <div className="item__link">
          <div className="link">
            <span className="link__placeholder">{text}</span>
            <div className="link__mask">
              <a className="link__text" href={url} target="_blank">{text}</a>
            </div>
          </div>
        </div>
      ) : (
        <div className="item__plain-text">
          <p className="">{text}</p>
        </div>
      )}
  </li>;

export default ListItem;
