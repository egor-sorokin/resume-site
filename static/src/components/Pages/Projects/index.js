import React from 'react';
import PropTypes from 'prop-types';

import { SWITCHER_PROJECTS } from '../../../constants/index';
import '../../../styles/app/_common.scss';
import './styles.scss';
import Switcher from '../../Switcher';
import DescriptionList from '../../DescriptionList';


const Projects = ({ projects }) => {
  const projectsComponent = projects.map(({
    id, name, info, image, url,
  }) => (
    <section key={id} className="slide-inner">
      <div className="container">
        <section className="content content--projects">
          <div className="col col--left">
            <h1 className="font-s-40">
              {name}
            </h1>
            <DescriptionList
              listData={info}
            />
          </div>
          <div className="col col--right">
            <img
              className="image"
              title={image.alt}
              alt={image.alt}
              src={image.src}
            />
          </div>
        </section>
      </div>
      <Switcher
        url={url}
        switcherLink={SWITCHER_PROJECTS}
      />
    </section>
  ));

  return (
    <main id="projects">
      {projectsComponent}
    </main>
  );
};


Projects.propTypes = {
  projects: PropTypes.instanceOf(Array),
};

Projects.defaultProps = {
  projects: [],
};


export default Projects;
