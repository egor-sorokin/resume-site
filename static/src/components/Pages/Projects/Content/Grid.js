import React from 'react';
import GridItem from './GridItem';
import './styles.scss';


const Grid = ({ data }) => {
  const gridComponent = data.map((item) =>
    <GridItem key={item.id}></GridItem>
  );

  return (
    <ul>
      {gridComponent}
    </ul>
  );
};


export default Grid;