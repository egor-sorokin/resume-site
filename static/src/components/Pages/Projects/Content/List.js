import React from 'react';
import ListItem from './ListItem';
import './styles.scss';


const List = ({ data }) => {
  const listComponent = data.map((item) =>
    <ListItem key={item.id}></ListItem>
  );

  return (
    <ul>
      {listComponent}
    </ul>
  );
};

export default List;
