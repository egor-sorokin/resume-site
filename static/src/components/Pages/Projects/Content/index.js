import React from 'react';
import Grid from './Grid'
import './styles.scss';


const Content = ({ data }) => (
  <section>
    <h1 class="title">{data.project}</h1>
    <ContentList></ContentList>
    {data.imgUrl ? (
        <div className="content content--right">
          <div className="image" style={{ backgroundImage: `url(${data.imgUrl})` }} />
        </div>
      ) : (
        <Grid></Grid>
      )}
  </section>
);

export default Content;
