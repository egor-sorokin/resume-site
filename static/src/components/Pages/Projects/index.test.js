// eslint-disable-next-line
import {shallow, render, mount} from 'enzyme';
import React from 'react';
import Projects from './index';
import Content from '../../Content/index';
import Switcher from '../../Switcher/index';


describe('<Projects />', () => {
  let component;
  const projects = [
    {
      id: 1,
      name: 'test name projects',
      url: 'test.url',
      info: [
        {
          label: 'test role',
          text: 'test text',
          url: 'test.info.url'
        }
      ],
      image: {
        src: 'test.src',
        alt: 'test alt'
      },
      experiments: []
    },
    {
      id: 2,
      name: 'test name experiments',
      url: '',
      info: [
        {
          label: 'test role 2',
          text: '-',
          url: ''
        }
      ],
      image: {
        src: '',
        alt: ''
      },
      experiments: [
        {
          id: 1,
          label: 'title',
          url: '',
          image: 'assets/exp_name.jpg'
        }
      ]
    }
  ];


  describe('base tests', () => {
    beforeEach(() => {
      component = shallow(<Projects />);
    });


    it('renders without crashing', () => {
      expect(component).toBeDefined();
    });


    it('matches snapshot', () => {
      expect(component).toMatchSnapshot();
    });


    it('renders 0 components', () => {
      expect(component.children()).toHaveLength(0);
    });
  });


  describe('props test', () => {
    it('renders data', () => {
      component = mount(<Projects />);
      component.setProps({data: {projects}});

      expect(component.prop('data').projects).toEqual(projects);
      // console.log(component.debug());
      expect(component.find('.slide-inner')).toHaveLength(2);
      expect(component.find('.line--top')).toHaveLength(2);
      expect(component.find('.line--bottom')).toHaveLength(1);
      expect(component.find('#projects')).toHaveLength(1);
      expect(component.find('#experiments')).toHaveLength(1);
      expect(component.find(<Content />)).toBeTruthy();
      expect(component.find(<Switcher />)).toBeTruthy();

      component.setProps({data: {}});
    });
  });
});
