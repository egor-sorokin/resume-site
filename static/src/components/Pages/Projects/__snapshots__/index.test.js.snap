// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Projects /> base tests matches snapshot 1`] = `<main />`;
